R/upkeep.R

```diff
@@ -16,30 +39,90 @@ use_tidy_upkeep_issue <- function(year = NULL) {
     }
   }
 
-  checklist <- upkeep_checklist(year)
+  checklist <- if (tidy) tidy_upkeep_checklist(year) else upkeep_checklist()
 
-  maybe_year <- if (is.null(year)) "" else glue(" ({year})")
+  title_year <- year %||% format(Sys.Date(), "%Y")
 
   gh <- gh_tr(tr)
   issue <- gh(
     "POST /repos/{owner}/{repo}/issues",
-    title = glue("Upkeep for {project_name()}{maybe_year}"),
+    title = glue("Upkeep for {project_name()} ({title_year})"),
     body = paste0(checklist, "\n", collapse = "")
   )
   Sys.sleep(1)
   view_url(issue$html_url)
 }
 
-upkeep_checklist <- function(year = NULL,
-                             posit_pkg = is_posit_pkg(),
-                             posit_person_ok = is_posit_person_canonical()) {
-  year <- year %||% 2000
+upkeep_checklist <- function() {
 
+  bullets <- c(
+    todo("`usethis::use_readme_rmd()`", !file_exists(proj_path("README.Rmd"))),
+    todo("`usethis::use_roxygen_md()`", !is_true(uses_roxygen_md())),
+    todo("`usethis::use_github_links()`", !has_github_links()),
+    todo("`usethis::use_pkgdown_github_pages()`", !uses_pkgdown()),
+    todo("
+        `usethis::use_package_doc()`.
+        Consider letting usethis manage your `@importFrom` directives here. \\
+        `usethis::use_import_from()` is handy for this.",
+        !has_package_doc()
+    ),
+    todo("
+         `usethis::use_testthat()`. \\
+         Learn more about testing at https://r-pkgs.org/tests.html",
+         !uses_testthat()
+    ),
+    todo("
+        `usethis::use_testthat(3)` and upgrade to 3e, \\
+        [testthat 3e vignette](https://testthat.r-lib.org/articles/third-edition.html)",
+        uses_old_testthat_edition(current = 3)
+    ),
+    todo("
+        Align the names of `R/` files and `test/` files for workflow happiness. \\
+        The docs for `usethis::use_r()` include a helpful script. \\
+        `usethis::rename_files()` may be be useful."),
+    todo("`usethis::use_github_action('check-standard')`", !uses_github_actions()),
+    todo(
+      "Consider changing default branch from `master` to `main`",
+      git_default_branch() == "master"
+    ),
+    todo("`usethis::use_code_of_conduct()`", !has_coc()),
+    todo(
+      "Modernize citation files; see `usethis::use_citation()`",
+      has_citation_file()
+    ),
+    todo("Remove check environments section from `cran-comments.md`"),
```

Comment:
I wonder if a typical community member will know what this is about?

Also, should it be conditional on `file_exists(proj_path("cran-comments.md"))`?

Perhaps, if `cran-comments.md` exists, we say something like: "Remove any description of check environments from `cran-comments.md`.

Reply:
Yeah, I like that - thanks!